pytensor/tensor/rewriting/math.py

```diff
@@ -362,9 +362,11 @@ def local_exp_log_nan_switch(fgraph, node):
         old_out = node.outputs[0]
         new_out = switch(le(x, 0), sub(1, exp(x)), np.asarray(np.nan, old_out.dtype))
         return [new_out]
-    
+
     # Case for log1mexp(log1mexp(x)) -> x
-    if isinstance(prev_op, aes_math.Log1mexp) and isinstance(node_op, aes_math.Log1mexp):
+    if isinstance(prev_op, aes_math.Log1mexp) and isinstance(
```

Comment:
This makes more sense as a separate rewrite, analogous to the `local_func_inv`, something like `local_func_inf_nan_switch`. Otherwise looks nice. Would also need a test